src/Authenticator/JwtAuthenticator.php

```diff
@@ -149,7 +155,18 @@ protected function decodeToken(string $token): ?object
     {
         $jsonWebKeySet = $this->getConfig('jwks');
         if ($jsonWebKeySet) {
-            $keySet = JWK::parseKeySet($jsonWebKeySet);
+            $jsonWebKeySetCache = $this->getConfig('jwksCache');
+            if ($jsonWebKeySetCache) {
+                $keySet = new CachedKeySet(
+                    $jsonWebKeySet,
+                    new Client(),
+                    new RequestFactory(),
+                    new Psr16Adapter(Cache::pool($jsonWebKeySetCache)),
```

Comment:
Given that `symfony/cache` is an optional dependency the existence of the `Psr16Adapter` class should be checked first and in case of failure show a message stating which package the dev should include in their app.